src/Parser/Shaman/Elemental/Modules/Items/EchoesOfTheGreatSundering.js

```diff
@@ -6,66 +6,118 @@ import ITEMS from 'common/ITEMS';
 import SPELLS from 'common/SPELLS';
 import ItemIcon from 'common/ItemIcon';
 import ItemLink from 'common/ItemLink';
-import { formatPercentage } from 'common/format';
 import ItemDamageDone from 'Main/ItemDamageDone';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import {formatPercentage} from 'common/format';
+import { ECHOES, CRIT_MULTIPLIER } from '../../Constants.js';
 
 class EchoesOfTheGreatSundering extends Analyzer {
-    static dependencies = {
-        combatants: Combatants,
-    };
+  static dependencies = {
+    combatants: Combatants,
+  };
 
-    buffedEarthquakeCasts = 0;
-    echoesProcsCounter = 0;
-    totalEarthquakeDamage = 0;
-    totalEarthquakeCasts = 0;
+  echoesProcsCounter = 0;
+  unbuffedBaseDamageSum = 0;
+  buffedBaseDamageSum = 0;
+  buffedEarthquakeDamage=0;
 
-    get estimate_bonus_damage() {
-        // Sorry for the magic number, but the goal is to estimate the bonus damage from buffed EQs by ~roughly~ calculating
-        // what fraction of EQ damage can be considered a result of the shoulder buff's 100% increase.
-        return  (0.5 * this.totalEarthquakeDamage * (this.buffedEarthquakeCasts / this.totalEarthquakeCasts)) || 0;
-    }
+  unbuffedTickCounter = 0;
+  buffedTickCounter = 0;
+  buffedCastCounter=0;
 
-    on_initialized() {
-        this.active = this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id);
-    }
+  state = 0;  //0=guaranteed not buffed; 1=guaranteed buffed; 2=not too sure(use heuristic)
+  endtime = 0;
 
-    on_byPlayer_cast(event) {
-        const spellId = event.ability.guid;
-        if (spellId !== SPELLS.EARTHQUAKE.id) {
-            return;
-        }
-        if (this.combatants.selected.hasBuff(SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id, event.timestamp)) {
-            this.buffedEarthquakeCasts += 1;
-        }
-        this.totalEarthquakeCasts += 1;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id) {
+      this.echoesProcsCounter += 1;
     }
+  }
 
-    on_byPlayer_damage(event) {
-        const spellId = event.ability.guid;
-        if (spellId === SPELLS.EARTHQUAKE_DAMAGE.id) {
-            this.totalEarthquakeDamage += event.amount;
-        }
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.EARTHQUAKE.id)
+      return;
+
+    if (this.combatants.selected.hasBuff(SPELLS.ECHOES_OF_THE_GREAT_SUNDERING_BUFF.id, event.timestamp)) {
+      this.buffedCastCounter++;
+      this.endtime = event.timestamp + ECHOES.PROC_DURATION + ECHOES.LAG_TOLERANCE;
+      this.state = 1;
```

Comment:
Please make constants at the top of your file for magic numbers to make it easier to find what they mean.